src/Microsoft.AspNetCore.Razor.Language/Legacy/CSharpCodeParser.cs

```diff
@@ -1576,166 +1572,198 @@ private void EnsureDirectiveIsAtStartOfLine()
 
         private void HandleDirective(DirectiveDescriptor descriptor)
         {
-            Context.Builder.CurrentBlock.Type = BlockKindInternal.Directive;
-            Context.Builder.CurrentBlock.ChunkGenerator = new DirectiveChunkGenerator(descriptor);
             AssertDirective(descriptor.Directive);
 
-            AcceptAndMoveNext();
-            Output(SpanKindInternal.MetaCode, AcceptedCharactersInternal.None);
+            var directiveErrorSink = new ErrorSink();
+            var savedErrorSink = Context.ErrorSink;
+            Context.ErrorSink = directiveErrorSink;
```

Comment:
Is there any significance of switching the error sink here? Looks like we are directly consuming it from `directiveErrorSink` at the end.